File: app/src/main/java/com/dev/cameronc/hues/Dagger/HueModule.java
package com.dev.cameronc.hues.Dagger;

import com.philips.lighting.hue.sdk.PHHueSDK;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by ccord on 11/9/2016.
 */

@Module
public class HueModule
{
    @Provides
    @Singleton
    public PHHueSDK provideHueSdk()
    {
        return PHHueSDK.create();
    }
}

File: app/src/main/java/com/dev/cameronc/hues/Dagger/HueModule.kt
package com.dev.cameronc.hues.Dagger

import android.os.Build
import com.philips.lighting.hue.sdk.PHHueSDK
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by ccord on 11/9/2016.
 */

@Module
class HueModule
{
    @Provides
    @Singleton
    fun provideHueSdk(): PHHueSDK
    {
        val hueSdk = PHHueSDK.create()
        hueSdk.deviceName = Build.MODEL
        hueSdk.appName = "Hues"

        return hueSdk
    }
}
